chain/postgresdb/pgdb/models.go

```diff
@@ -0,0 +1,15 @@
+package pgdb
+
+import "github.com/drand/drand/chain"
+
+// dbBeacon represents a beacon that is stored in the database.
+type dbBeacon struct {
+	PreviousSig []byte `db:"previous_sig"`
+	Round       uint64 `db:"round"`
+	Signature   []byte `db:"signature"`
+}
+
+func toChainBeacon(dbB dbBeacon) *chain.Beacon {
+	cb := chain.Beacon(dbB)
```

Comment:
very random noob gopher question:

is there any difference in allocation between:

```
cb := chain.Beacon(dbB)
return &cb
```
and

```
cb := &chain.Beacon(dbB)
return cb
```
and

```
return &chain.Beacon(dbB)
```

my google-fu gave me some loose info about escape analysis when using pointers generally, but wasn't clear on specifics for these scenarios

Reply:
The compiler will optimize this anyway, so from a compiler perspective, there's no difference.

However, this is a bit more useful as it helps with the idea of using pointers when needed and values everywhere else.

Using values is preferred at it results in less race condition issues. However, when sharing the same memory address is required, then pointers should be used diligently.

I think @ardan-bkennedy can provide more details about this.

Reply:
I teach to always construct values using value semantics when assigning to a variable and use the & to show that the value is being shared. I think this provides better readability.

_Make things easy to understand, not easy to do._

This will not confused the compiler with the escape analysis, so it's better to go for readability.

If you can do the construction on the return, and it's readable, I would go with that option since it's less lines of code.

Reply:
This could be one line of code on the return

Reply:
Ty for the tips!